src/Security/RolesMatrixBuilder.php

```diff
@@ -0,0 +1,302 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Security;
+
+use Sonata\AdminBundle\Admin\Pool;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * @author Silas Joisten <silasjoisten@hotmail.de>
+ */
+class RolesMatrixBuilder
+{
+    /**
+     * @var TokenStorageInterface
+     */
+    protected $tokenStorage;
+
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    protected $authorizationChecker;
+
+    /**
+     * @var Pool
+     */
+    protected $pool;
+
+    /**
+     * @var TranslatorInterface
+     */
+    protected $translator;
+
+    /**
+     * @var array
+     */
+    protected $rolesHierarchy;
+
+    /**
+     * @var array
+     */
+    private $labelPermission;
+
+    /**
+     * @var array
+     */
+    private $labelAdmin;
+
+    /**
+     * @var array
+     */
+    private $exclude;
+
+    public function __construct(
+        TokenStorageInterface $tokenStorage,
+        AuthorizationCheckerInterface $authorizationChecker,
+        Pool $pool,
+        array $rolesHierarchy = []
+    ) {
+        $this->tokenStorage = $tokenStorage;
+        $this->authorizationChecker = $authorizationChecker;
+        $this->pool = $pool;
+        $this->rolesHierarchy = $rolesHierarchy;
+        $this->labelPermission = [];
```

Comment:
@jordisala1991 @kunicmarko20 shall we remove this and use ´protected labelPermission = [];´ instead?

Reply:
Yes